Source/UI/PaymentForm/ViewModel/DefaultPaymentViewModel.swift

```diff
@@ -175,6 +174,45 @@ extension DefaultPaymentViewModel: PaymentViewControllerDelegate {
 
   func cardholderIsUpdated(value: String) {
     cardDetails.name = value
+    validateMandatoryInputProvided()
+  }
+
+  private func validateMandatoryInputProvided() {
+    var isMandatoryInputProvided = false
+    defer { shouldEnablePayButton?(isMandatoryInputProvided) }
+
+    // If a scheme is not recorded the number hasn't started being inputted
+    // so we can safely know mandatory fields are not provided
+    guard let cardScheme = cardDetails.scheme else { return }
+
+    // Check if cardholder is required and if so whether it is provided
+    let isCardholderRequired = paymentFormStyle?.cardholderInput?.isMandatory == true
+    if isCardholderRequired && cardDetails.name.isEmpty { return }
+
+    // Check if security code is displayed and if so whether it is valid
+    // This is business logic that wants Security code to be mandatory whenever its shown
+    let isSecurityCodeRequired = paymentFormStyle?.securityCode != nil
+    if isSecurityCodeRequired && !cardValidator.isValid(cvv: cardDetails.cvv, for: cardScheme) { return }
+
+    // Check if Billing is required and if so whether it exists
+    let isAddBillingRequired = paymentFormStyle?.addBillingSummary?.isMandatory == true
+    let isEditBillingRequired = paymentFormStyle?.editBillingSummary?.isMandatory == true
+    let isBillingRequired = isAddBillingRequired && isEditBillingRequired
+    if isBillingRequired && cardDetails.billingAddress == nil { return }
```

Comment:
```suggestion
      guard paymentFormStyle?.addBillingSummary?.isMandatory == false,
            paymentFormStyle?.editBillingSummary?.isMandatory == false,
            cardDetails.billingAddress != nil else { return }
```

Reply:
Your refactor suggestion doesn't do the same thing leaving early for nil. 

This is exactly why the original approach is trying to improve clarity by breaking complex logic down to readable logic.